support-frontend/assets/helpers/abTests/abtest.js

```diff
@@ -201,6 +202,15 @@ function getParticipations(
       return;
     }
 
+    if (test.urlParticipationOnly) {
+      const participationsFromUrl = getParticipationsFromUrl();
+      if (participationsFromUrl && participationsFromUrl[testId]) {
+        participations[testId] = participationsFromUrl[testId];
+      } else {
+        return;
```

Comment:
so if any test is url participation and its not in the url, we would return undefined? (not in any test?)

Reply:
we're inside a `forEach` here so it just abandons this particular test.

Reply:
A nicer and more functional approach would be to use `reduce`, though it would still be mutating an object as it goes

Reply:
did not know return returns from a foreach 🔢